File: crazy-random-multiverse-server/src/main/java/com/dainws/games/crm/stomp/dto/models/ZoneDto.java
package com.dainws.games.crm.stomp.dto.models;

import jakarta.validation.constraints.NotNull;

public class ZoneDto {
	@NotNull
	private PlayerDto owner;
	
	@NotNull
	private CardDto[][] combatants;
	
	public void setOwner(PlayerDto owner) {
		this.owner = owner;
	}
	
	public PlayerDto getOwner() {
		return owner;
	}
	
	public void setCombatants(CardDto[][] combatants) {
		this.combatants = combatants;
	}
	
	public CardDto[][] getCombatants() {
		return combatants;
	}
}

File: crm-view-web/src/infrastructure/stomp/dto/ZoneDto.ts
import PlayerDto from '@/infrastructure/stomp/dto/PlayerDto';
import CardDto from '@/infrastructure/stomp/dto/CardDto';

type ZoneDto = {
  owner: PlayerDto;
  health: number;
  maxHealth: number;
  combatants: CardDto[][];
};

export default ZoneDto;
